src/lib/mdns/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -209,36 +216,80 @@ CHIP_ERROR AdvertiserMinMdns::Start(chip::Inet::InetLayer * inetLayer, uint16_t
 /// Stops the advertiser.
 CHIP_ERROR AdvertiserMinMdns::StopPublishDevice()
 {
-    mQueryResponderAllocatorOperational.Clear();
+    for (auto & allocator : mQueryResponderAllocatorOperational)
+    {
+        allocator.Clear();
+    }
     mQueryResponderAllocatorCommissionable.Clear();
     mQueryResponderAllocatorCommissioner.Clear();
     return CHIP_NO_ERROR;
 }
 
+QueryResponderAllocator<AdvertiserMinMdns::kMaxOperationalRecords> *
+AdvertiserMinMdns::FindOperationalAllocator(const FullQName & qname)
+{
+    for (auto & allocator : mQueryResponderAllocatorOperational)
+    {
+        if (allocator.GetResponder(QType::SRV, qname) != nullptr)
+        {
+            return &allocator;
+        }
+    }
+    return nullptr;
+}
+
+QueryResponderAllocator<AdvertiserMinMdns::kMaxOperationalRecords> * AdvertiserMinMdns::FindEmptyOperationalAllocator()
+{
+    for (auto & allocator : mQueryResponderAllocatorOperational)
+    {
+        if (allocator.IsEmpty())
+        {
+            return &allocator;
+        }
+    }
+    return nullptr;
+}
+
 CHIP_ERROR AdvertiserMinMdns::Advertise(const OperationalAdvertisingParameters & params)
 {
-    // TODO: When multi-admin is enabled, commissionable does not need to be cleared here.
-    mQueryResponderAllocatorOperational.Clear();
-    char nameBuffer[64] = "";
+    char nameBuffer[kOperationalServiceNamePrefix + 1] = "";
 
     /// need to set server name
     ReturnErrorOnFailure(MakeInstanceName(nameBuffer, sizeof(nameBuffer), params.GetPeerId()));
 
+    char nameCheckStorage[kMaxOperationalServiceNameSize + 4 * sizeof(QNamePart)];
```

Comment:
a comment of why 4*pointer size would help reading the code. Like RequiredStorageSize cannot be used and kMaxNameSize includes strlen + 1 part. 

When reading the code equivalent, I wonder if the '+1' that is used by the required size methods is included in the max size here.

Reply:
Now that you've drawn my attention to it, FlatAllocatedQName is overkill here - we don't need to actually copy the parts into that buffer because nameBuffer doesn't go out of scope before the nameCheck. I can just use an array of QNameParts (const char*) to instantiate the QName and it's fine. Removed the storage completely.